File: javascript/matrix/matrix.js
const range = n => [...Array(n).keys()];

const extractColumnValue = (index) =>
      (row) => row[index];

export class Matrix {

  constructor(rawInput) {
    this.rawInput = rawInput;
  }

  get rows() {
    return this.rawInput.split("\n").map(
      rowstr => rowstr.split(' ').map(Number));
  }

  get columns() {
    return range(this.width)
      .map(rowIndex =>
           this.rows.map(
             extractColumnValue(rowIndex)));
  }

  get width() {
    return Math.max(
      ...this.rows.map(row => row.length));
  }
}

File: python/matrix/matrix.py
class Matrix(object):
    def __init__(self, matrix_string):
        lines = matrix_string.split("\n")
        self.rows = list(map(
            lambda line: list(map(int, line.split())),
            lines))

    def row(self, index):
        return self.rows[index-1]

    def column(self, index):
        return list(map(lambda row: row[index-1], self.rows))
